### ISSUE TITLE
Better documentation for "supports_pattern"

### ISSUE THREAD

**Issue by markstos** [qutebrowser/qutebrowser#6454]
I was reading the docs for `content.notifications.show_origin` and I see that valid values are of type `bool`. But the description refers to host pattern support:

_Note that URL patterns with this setting only get matched against the origin part of the URL_

Further searching lead to an undocumented setting in a YAML file that says "supports_pattern" is "true" for this setting.

So there does appear to be kind of host-pattern support for some settings, but it's not clear how it works. Here's some suggestion for how to make this clearer:

Where docs say "_This setting supports URL patterns._", the words "URL patterns" can link to a section in the docs that describes how URL patterns work and provide an example, because it's not clear how a setting of type boolean can support URL patterns.

**Comment by The-Compiler** [comment]
The `supports_pattern` in `configdata.yml` is internal to qutebrowser, so there's no need to document that one to users.

How to use URL patterns is documented though, both in the [documentation for `:set`](https://qutebrowser.org/doc/help/commands.html#set) as well as the general configuration docs ([via user interface](https://github.com/qutebrowser/qutebrowser/blob/master/doc/help/configuring.asciidoc#configuring-qutebrowser-via-the-user-interface), [via config.py](https://github.com/qutebrowser/qutebrowser/blob/master/doc/help/configuring.asciidoc#per-domain-settings)).

Seems like a good idea to link to one of those, though.

**Comment by markstos** [comment]
Thanks. Here some related suggestions;

  - It would be helpful if there was a section at the top of `help/settings.html` that either explained how to set the settings or linked to those docs.
  - It's not clear how the "settings" docs relate to "configuring" docs. Since I was focused on trying to find document for a "setting" it didn't occur to me to look at the other "Configuring" page. I see now they are really referring to the mostly same thing.
  
To address those suggestions and because the "Configuring" page is fairly short and closely related to "settings.html", I recommend merging the two pages into one. At the top, have the "configuring" content. There can be a link "Jump to list of all settings", much like recipe blogs now include "jump to recipe" while still including a bunch of opening prose who people who want it. 

The table of contents page can continue to link to both points in the document as it does now. 

The language of "Configuring" and "Settings" could also be better aligned. When referring to updating settings, just use subheads like "Changing settings via the user interface".  Use "Configuring" when it's clearly a different use than "changing settings". 
 
If the language was consistent or the docs were closer together, users in situations like mine would have had a better chance of finding the "settings" docs they were looking for and not opening another issue. :)

**Comment by The-Compiler** [comment]
> To address those suggestions and because the "Configuring" page is fairly short and closely related to "settings.html", I recommend merging the two pages into one.

I disagree - I wouldn't call it "fairly short", and I see it as a best practice to keep reference-like documentation and "how-to" documentation separate (also see [Divio's page on the topic](https://documentation.divio.com/)). If someone wants to learn how qutebrowser can be configured in general, they likely don't care about what each of the >300 settings does. Similarly, if someone wants to know something about a setting they're looking at, they likely already figured out how to set that setting.

Seems like a good idea to link them to each other. Also I'd really like to have a more structured documentation in general, but that's currently blocked on appropriate tooling (#345), which I've never gotten around to finishing unfortunately.

> The language of "Configuring" and "Settings" could also be better aligned. When referring to updating settings, just use subheads like "Changing settings via the user interface". Use "Configuring" when it's clearly a different use than "changing settings".

I don't see what's inconsistent about it. "Configuring qutebrowser" refers to setting settings and to binding keys. The configuring page (as well as those headings) explain both.
